examples/rcbd_one_var.R

```diff
@@ -0,0 +1,41 @@
+wheat.data <- read.csv("Wheat_yield_data.csv", header = TRUE, stringsAsFactors = FALSE)
+
+library(car)  # for leveneTest()
+library(agricolae)  # for LSD.test()
+
+#Inform R that Block and Treatment are factors
+wheat.data$Block<-as.factor(wheat.data$Block)
+wheat.data$Treatment<-as.factor(wheat.data$Treatment)
+
+##The ANOVA##
+wheat.mod<-lm(Yield ~ Treatment + Block, wheat.data)
+anova(wheat.mod)
+
+##TESTING ASSUMPTIONS##
+#Generate residual and predicted values
+wheat.data$resids <- residuals(wheat.mod)
+wheat.data$preds <- predict(wheat.mod)
+wheat.data$sq_preds <- wheat.data$preds^2
+
+#Look at a plot of residual vs. predicted values
+plot(resids ~ preds, data = wheat.data,
+     xlab = "Predicted Values", ylab = "Residuals")
+#Create Box Plot of treatments
+boxplot(Yield ~ Treatment, data = wheat.data,
+        main = "Effect of wheat variety on yield",
+        xlab = "Wheat cultivar", ylab = "Yield (ton/ha)")
+#Perform a Shapiro-Wilk test for normality of residuals
+shapiro.test(wheat.data$resids)
+qqnorm(resid(wheat.mod))
+plot(density(resid(wheat.mod)))
+
+#Perform Levene's Test for homogenity of variances among treatments
+leveneTest(Yield ~ Treatment, data = wheat.data)
```

Comment:
Great question. I have gotten 2 different perspectives on this, but I am going with what is taught in PLS205 at UC Davis and at Univ of NH (http://www.unh.edu/halelab/BIOL933/labs/lab5.pdf):
"For RCBD's, if you intend to make mean comparisons among treatment levels, you must do
Levene's Test for treatment, using a one-way ANOVA. Levene's Test is valid only for
one-way ANOVA's." <== Thus, they remove the other factors, and run the Levene's test for each factor that they are interested in. Since we assume they are not interested in "Block", there is not test for that. However, when we do the code for the other scenarios with multiple treatments, we will need to have a Levene's test for each one.